OutOfSchool/OutOfSchool.BusinessLogic/Services/CurrentUserService.cs

```diff
@@ -295,43 +297,38 @@ private async Task<bool> EmployeeWorkshopRights(Guid workshopId)
         return isUserRelatedEmployee;
     }
 
-    private Task<bool> ModeratorHasRights(Guid moderatorId)
+    private async Task<bool> ModeratorHasRights()
     {
         if (!IsModerator())
         {
-            return Task.FromResult(false);
+            return false;
         }
 
-        var result = UserId == moderatorId.ToString();
+        var result = await moderatorRepository.Any(m => m.Individual.UserId == UserId);
```

Comment:
Technically, moderator/techadmin `Id` is both a primary key to moderator/techadmin as well as foreign key to individual.
As this is not the issue now, but we can add individual `Id` to token during auth and just do the `Any(m => m.Id = <claim_from_toekn>)`